### ISSUE TITLE
bug: default value for Array field would corrupt the table

### ISSUE THREAD

**Issue by everpcpc** [datafuselabs/databend#9260]
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

v0.8.150-nightly

### What's Wrong?

insert into a table with default value for array would result in error:

```
ERROR 1105 (HY000): result row write failed: Code: 1104, displayText = called `Option::unwrap()` on a `None` value.
```

### How to Reproduce?

```sql
CREATE TABLE `data2` (
  `i64` BIGINT,
  `u64` BIGINT UNSIGNED,
  `f64` DOUBLE,
  `s` VARCHAR,
  `s2` VARCHAR,
  `a16` ARRAY(INT16),
  `a8` ARRAY(UINT8),
  `d` DATE,
  `t` TIMESTAMP
);

insert into data2 (s, s2) values ('abc', '123');

select * from data2;
```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by BohuTANG** [comment]
cc @sundy-li

**Comment by sundy-li** [comment]
can't reproduce in the latest version:
```

MySQL [(none)]> CREATE TABLE `data2` (
    ->   `i64` BIGINT,
    ->   `u64` BIGINT UNSIGNED,
    ->   `f64` DOUBLE,
    ->   `s` VARCHAR,
    ->   `s2` VARCHAR,
    ->   `a16` ARRAY(INT16),
    ->   `a8` ARRAY(UINT8),
    ->   `d` DATE,
    ->   `t` TIMESTAMP
    -> );
Query OK, 0 rows affected (0.006 sec)
MySQL [(none)]> insert into data2 (s, s2) values ('abc', '123');
Query OK, 1 row affected (0.006 sec)

MySQL [(none)]> select * from data2;
+------+------+------+------+------+------+------+------------+----------------------------+
| i64  | u64  | f64  | s    | s2   | a16  | a8   | d          | t                          |
+------+------+------+------+------+------+------+------------+----------------------------+
|    0 |    0 |  0.0 | abc  | 123  | []   | []   | 1970-01-01 | 1970-01-01 00:00:00.000000 |
+------+------+------+------+------+------+------+------------+----------------------------+
1 row in set (0.007 sec)

```

**Comment by everpcpc** [comment]
Wired, I still can reproduce this on my local env.
The panic log:

```
2022-12-19T10:26:39.105773Z ERROR common_tracing::panic_hook: src/common/tracing/src/panic_hook.rs:38: panicked at 'called `Option::unwrap()` on a `None` value', /Users/everpcpc/.cargo/git/checkouts/arrow2-8a2ad61d
97265680/1da33ac/src/io/parquet/read/row_group.rs:69:37 backtrace=Backtrace [{ fn: "std::backtrace_rs::backtrace::libunwind::trace", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/../../back
trace/src/backtrace/libunwind.rs", line: 93 }, { fn: "std::backtrace_rs::backtrace::trace_unsynchronized", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/../../backtrace/src/backtrace/mod.rs
", line: 66 }, { fn: "std::backtrace::Backtrace::create", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/backtrace.rs", line: 332 }, { fn: "common_tracing::panic_hook::log_panic", file: "./s
rc/common/tracing/src/panic_hook.rs", line: 35 }, { fn: "common_tracing::panic_hook::set_panic_hook::{{closure}}", file: "./src/common/tracing/src/panic_hook.rs", line: 30 }, { fn: "<alloc::boxed::Box<F,A> as core:
:ops::function::Fn<Args>>::call", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/alloc/src/boxed.rs", line: 2032 }, { fn: "std::panicking::rust_panic_with_hook", file: "/rustc/e631891f7ad40eac3ef58e
c3c2b57ecd81e40615/library/std/src/panicking.rs", line: 692 }, { fn: "std::panicking::begin_panic_handler::{{closure}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panicking.rs", line: 5
77 }, { fn: "std::sys_common::backtrace::__rust_end_short_backtrace", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/sys_common/backtrace.rs", line: 137 }, { fn: "rust_begin_unwind", file: "
/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panicking.rs", line: 575 }, { fn: "core::panicking::panic_fmt", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/panicking.rs",
 line: 65 }, { fn: "core::panicking::panic", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/panicking.rs", line: 115 }, { fn: "core::option::Option<T>::unwrap", file: "/rustc/e631891f7ad40e
ac3ef58ec3c2b57ecd81e40615/library/core/src/option.rs", line: 778 }, { fn: "<arrow2::io::parquet::read::row_group::RowGroupDeserializer as core::iter::traits::iterator::Iterator>::next::{{closure}}", file: "/Users/
everpcpc/.cargo/git/checkouts/arrow2-8a2ad61d97265680/1da33ac/src/io/parquet/read/row_group.rs", line: 69 }, { fn: "core::iter::adapters::map::map_try_fold::{{closure}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57
ecd81e40615/library/core/src/iter/adapters/map.rs", line: 91 }, { fn: "core::iter::traits::iterator::Iterator::try_fold", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/iter/traits/iterator
.rs", line: 2238 }, { fn: "<core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::try_fold", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/iter/adapters/map.rs",
line: 117 }, { fn: "<core::iter::adapters::GenericShunt<I,R> as core::iter::traits::iterator::Iterator>::try_fold", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/iter/adapters/mod.rs", lin
e: 195 }, { fn: "core::iter::traits::iterator::Iterator::try_for_each", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/iter/traits/iterator.rs", line: 2299 }, { fn: "<core::iter::adapters::
GenericShunt<I,R> as core::iter::traits::iterator::Iterator>::next", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/iter/adapters/mod.rs", line: 178 }, { fn: "alloc::vec::Vec<T,A>::extend_d
esugared", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/alloc/src/vec/mod.rs", line: 2857 }, { fn: "<alloc::vec::Vec<T,A> as alloc::vec::spec_extend::SpecExtend<T,I>>::spec_extend", file: "/rustc/
e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/alloc/src/vec/spec_extend.rs", line: 18 }, { fn: "<alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter", file: "/rustc/e6318
91f7ad40eac3ef58ec3c2b57ecd81e40615/library/alloc/src/vec/spec_from_iter_nested.rs", line: 43 }, { fn: "<alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter", file: "/rustc/e631891f7ad40
eac3ef58ec3c2b57ecd81e40615/library/alloc/src/vec/spec_from_iter.rs", line: 33 }, { fn: "<alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57
ecd81e40615/library/alloc/src/vec/mod.rs", line: 2757 }, { fn: "core::iter::traits::iterator::Iterator::collect", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/iter/traits/iterator.rs", li
ne: 1836 }, { fn: "<core::result::Result<V,E> as core::iter::traits::collect::FromIterator<core::result::Result<A,E>>>::from_iter::{{closure}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/s
rc/result.rs", line: 2075 }, { fn: "core::iter::adapters::try_process", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/iter/adapters/mod.rs", line: 164 }, { fn: "<core::result::Result<V,E>
as core::iter::traits::collect::FromIterator<core::result::Result<A,E>>>::from_iter", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/result.rs", line: 2075 }, { fn: "core::iter::traits::ite
rator::Iterator::collect", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/iter/traits/iterator.rs", line: 1836 }, { fn: "<arrow2::io::parquet::read::row_group::RowGroupDeserializer as core:
:iter::traits::iterator::Iterator>::next", file: "/Users/everpcpc/.cargo/git/checkouts/arrow2-8a2ad61d97265680/1da33ac/src/io/parquet/read/row_group.rs", line: 66 }, { fn: "common_storages_fuse::io::read::block_rea
der::BlockReader::try_next_block", file: "./src/query/storages/fuse/fuse/src/io/read/block_reader.rs", line: 403 }, { fn: "common_storages_fuse::io::read::block_reader::BlockReader::deserialize", file: "./src/query
/storages/fuse/fuse/src/io/read/block_reader.rs", line: 325 }, { fn: "<common_storages_fuse::operations::fuse_source::FuseTableSource as common_pipeline_core::processors::processor::Processor>::process", file: "./s
rc/query/storages/fuse/fuse/src/operations/fuse_source.rs", line: 203 }, { fn: "common_pipeline_core::processors::processor::ProcessorPtr::process", file: "./src/query/pipeline/core/src/processors/processor.rs", li
ne: 109 }, { fn: "databend_query::pipelines::executor::executor_worker_context::ExecutorWorkerContext::execute_sync_task", file: "./src/query/service/src/pipelines/executor/executor_worker_context.rs", line: 90 },
{ fn: "databend_query::pipelines::executor::executor_worker_context::ExecutorWorkerContext::execute_task", file: "./src/query/service/src/pipelines/executor/executor_worker_context.rs", line: 80 }, { fn: "databend_
query::pipelines::executor::pipeline_executor::PipelineExecutor::execute_single_thread", file: "./src/query/service/src/pipelines/executor/pipeline_executor.rs", line: 300 }, { fn: "databend_query::pipelines::execu
tor::pipeline_executor::PipelineExecutor::execute_threads::{{closure}}::{{closure}}", file: "./src/query/service/src/pipelines/executor/pipeline_executor.rs", line: 256 }, { fn: "<core::panic::unwind_safe::AssertUn
windSafe<F> as core::ops::function::FnOnce<()>>::call_once", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/panic/unwind_safe.rs", line: 271 }, { fn: "std::panicking::try::do_call", file: "
/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panicking.rs", line: 483 }, { fn: "___rust_try" }, { fn: "std::panicking::try", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src
/panicking.rs", line: 447 }, { fn: "std::panic::catch_unwind", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panic.rs", line: 137 }, { fn: "common_base::runtime::catch_unwind::catch_unwind"
, file: "./src/common/base/src/runtime/catch_unwind.rs", line: 26 }, { fn: "databend_query::pipelines::executor::pipeline_executor::PipelineExecutor::execute_threads::{{closure}}", file: "./src/query/service/src/pi
pelines/executor/pipeline_executor.rs", line: 255 }, { fn: "common_base::runtime::thread::Thread::named_spawn::{{closure}}", file: "./src/common/base/src/runtime/thread.rs", line: 81 }, { fn: "std::sys_common::back
trace::__rust_begin_short_backtrace", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/sys_common/backtrace.rs", line: 121 }, { fn: "std::thread::Builder::spawn_unchecked_::{{closure}}::{{clos
ure}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/thread/mod.rs", line: 551 }, { fn: "<core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once", fil
e: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/panic/unwind_safe.rs", line: 271 }, { fn: "std::panicking::try::do_call", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/
panicking.rs", line: 483 }, { fn: "___rust_try" }, { fn: "std::panicking::try", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panicking.rs", line: 447 }, { fn: "std::panic::catch_unwind", f
ile: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panic.rs", line: 137 }, { fn: "std::thread::Builder::spawn_unchecked_::{{closure}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/libr
ary/std/src/thread/mod.rs", line: 550 }, { fn: "core::ops::function::FnOnce::call_once{{vtable.shim}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/ops/function.rs", line: 510 }, { fn: "
<alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/alloc/src/boxed.rs", line: 2000 }, { fn: "<alloc::boxed::Box<F,A> as core::o
ps::function::FnOnce<Args>>::call_once", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/alloc/src/boxed.rs", line: 2000 }, { fn: "std::sys::unix::thread::Thread::new::thread_start", file: "/rustc/e6
31891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/sys/unix/thread.rs", line: 108 }, { fn: "__pthread_deallocate" }] panic.file="/Users/everpcpc/.cargo/git/checkouts/arrow2-8a2ad61d97265680/1da33ac/src/io/parqu
et/read/row_group.rs" panic.line=69 panic.column=37
```

**Comment by sundy-li** [comment]
Ok, it may be fixed by #9279 of arrow2's new version.  Let's create a test after #9279  is merged.
